apps/sms/services/js/message_manager.js

```diff
@@ -510,9 +510,6 @@ var MessageManager = {
         _isMessageBelongTo1to1Conversation(number, message);
       if (isMessageInThread) {
         threadId = message.threadId;
-        // we need to set the current threadId,
-        // because we start sms app in a new window
-        Threads.registerMessage(message);
```

Comment:
"ensureThreadRegistered" is useful mainly to get the thread known -- and currently we do this through registerMessage which is maybe suboptimal.